8813 opex speed up cypress tests in deployed environments (formerly flexion#10478)

AnthonyLoeraTaxCourt opened the pull request.

ChristieLaferriereTaxCourt committed:
adding judge to test

ChristieLaferriereTaxCourt committed:
10478: speed up cypress tests by removing obsolete tests, async lambdas are no longer auto retried

AnthonyLoeraTaxCourt committed:
10478: Added bail flag, and changed a test to fail

ChristieLaferriereTaxCourt committed:
revert config.yml and court-issued-documents.cy.ts

AnthonyLoeraTaxCourt committed:
10478: Added spacing at end of config.yml file

AnthonyLoeraTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

AnthonyLoeraTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

ChristieLaferriereTaxCourt committed:
add cypress afterEach cleanup

ChristieLaferriereTaxCourt committed:
Merge branch '10478-opex-speed-up-cypress-tests-in-deployed-environmnts' of github.com:ustaxcourt/ef-cms into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

paulo-filho-ctr commented on cypress/local-only/support/index.ts:
```diff
@@ -3,11 +3,18 @@ import '@cypress/puppeteer/support';
+// before(() => {
+//   // Skip subsequent tests in spec when one fails.
+//   (cy.state('runnable').ctx as Mocha.Context).currentTest.parent.bail(true);
+// });
```
Should we keep the comments or it is ok to remove the delete the block altogether?

AnthonyLoeraTaxCourt replied:
We are just running test this is not the final pr. @paulo-filho-ctr

AnthonyLoeraTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

AnthonyLoeraTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

AnthonyLoeraTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

AnthonyLoeraTaxCourt committed:
10478: Reverted fail-fast tests, added new fail fast tests, edited run-cypress.sh file to handle fail fast method

AnthonyLoeraTaxCourt committed:
10478: Reverted fail-fast tests, added new fail fast tests, edited run-cypress.sh file to handle fail fast method

AnthonyLoeraTaxCourt committed:
10478: reverted puroseful failing test, reverted unneeded changes

AnthonyLoeraTaxCourt committed:
10478: Added failing test in smoketest

AnthonyLoeraTaxCourt committed:
modify logout to fail

AnthonyLoeraTaxCourt committed:
10478: Added cypress plugin

AnthonyLoeraTaxCourt committed:
10478: Added cypress plugin

ChristieLaferriereTaxCourt committed:
add return config

ChristieLaferriereTaxCourt committed:
remove conflicting terraform command

AnthonyLoeraTaxCourt committed:
10478: Edited env variable values, and configuration for parallel runs

AnthonyLoeraTaxCourt committed:
10478: Final test for fail fast method

AnthonyLoeraTaxCourt committed:
10478: Final Final test

AnthonyLoeraTaxCourt committed:
10478: Added script to run and clean flags created

AnthonyLoeraTaxCourt committed:
10478: EDited test, and added more configuration

AnthonyLoeraTaxCourt committed:
10478: moved it to a before each function

AnthonyLoeraTaxCourt committed:
10478: Code clean up

AnthonyLoeraTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

AnthonyLoeraTaxCourt committed:
10478: reverted fail-fast feature

AnthonyLoeraTaxCourt committed:
10478: added delay, increased parallelism, and experimental modifier

AnthonyLoeraTaxCourt committed:
10478: reverted delays and experimental so we only increase parallel runs

AnthonyLoeraTaxCourt committed:
10478: removed more delays

AnthonyLoeraTaxCourt committed:
10478: removed empty curly brackets

ChristieLaferriereTaxCourt committed:
test for reduced time with p-limit in user creation script

ChristieLaferriereTaxCourt committed:
test user creation script refactor

ChristieLaferriereTaxCourt committed:
refactor test user creation script to use p-limit and batch parrallel requests

ChristieLaferriereTaxCourt committed:
Merge branch '10478-opex-speed-up-cypress-tests-in-deployed-environmnts' of github.com:ustaxcourt/ef-cms into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

AnthonyLoeraTaxCourt committed:
10478: added docker image caching

paulo-filho-ctr committed:
10478: add multi thread to cypress jobs

paulo-filho-ctr committed:
Revert "10478: add multi thread to cypress jobs"

This reverts commit 3bdd2513964c966082b649a07f95181f32c6dcce.

paulo-filho-ctr committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

ChristieLaferriereTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environments

ChristieLaferriereTaxCourt committed:
Merge branch '10478-opex-speed-up-cypress-tests-in-deployed-environmnts' of github.com:ustaxcourt/ef-cms into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

ChristieLaferriereTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

codyseibert committed:
removing dynamo code for deleting old users

ChristieLaferriereTaxCourt committed:
changing back to 6 to test

codyseibert committed:
restructure tests to delete users after all tests run

codyseibert committed:
improve cleanup script

codyseibert committed:
8813: putting the cypress cleanup at the end

codyseibert committed:
improving the clean up script

codyseibert committed:
working on improving the clean up script

codyseibert committed:
adding comments

codyseibert committed:
making the username for irs unique to prevent collision

codyseibert committed:
fix issue with length of the test user

codyseibert committed:
remove extra -deploy

codyseibert committed:
Merge pull request #9118 from ustaxcourt/10478-opex-speed-up-cypress-tests-in-deployed-environmnts-updates

10478 opex speed up cypress tests in deployed environmnts updates

codyseibert commented on .circleci/config.yml:
```diff
@@ -378,14 +384,17 @@ jobs:
     resource_class: large
     steps:
+      - setup_remote_docker:
+          version: docker23
```
did this decrease the deploy times? and by how much?

codyseibert replied:
an update, it does seem like this speeds up the build by a few minutes

## without setup_remote_docker:

<img width="1409" height="140" alt="Screenshot 2025-09-10 at 11 27 01 AM" src="https://github.com/user-attachments/assets/58154526-c09d-49da-ac28-7cb1cba5dba1" />

## after setup_remote_docker:

<img width="1414" height="140" alt="Screenshot 2025-09-10 at 11 43 10 AM" src="https://github.com/user-attachments/assets/4a119fa8-29fa-4dd9-88f2-44d7959f548d" />

codyseibert committed:
10478: remove extra import

codyseibert committed:
10478: adding back p-queue I accidently deleted

codyseibert committed:
10478: down grade p-queue

codyseibert committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

ChristieLaferriereTaxCourt committed:
replace cimg/node:22.16.0 with custom efcms-docker-image to resolve permission issues with cache restoration

ChristieLaferriereTaxCourt committed:
removing install aws clu step since cache successfully restore

ChristieLaferriereTaxCourt committed:
Docker Image with AWS CLI Pre-installed

ChristieLaferriereTaxCourt committed:
retsore previous image

ChristieLaferriereTaxCourt committed:
restoring previous config.yml without updating docker image

ChristieLaferriereTaxCourt committed:
switching from 6 to 8 parallel runs

ChristieLaferriereTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

ChristieLaferriereTaxCourt committed:
add dependency cache

ChristieLaferriereTaxCourt committed:
Merge branch '10478-opex-speed-up-cypress-tests-in-deployed-environmnts' of github.com:ustaxcourt/ef-cms into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

ChristieLaferriereTaxCourt committed:
testing with just adding more runners

ChristieLaferriereTaxCourt committed:
adding retrieve from cache dependency step

ChristieLaferriereTaxCourt committed:
adding an id for cache so the install node dependencies step can validate cache existence before installing

ChristieLaferriereTaxCourt committed:
cache entire node modules instead

ChristieLaferriereTaxCourt committed:
cache the cypress binary

ChristieLaferriereTaxCourt committed:
adding updated cypress binary cache path

ChristieLaferriereTaxCourt committed:
check cache path for cypress binary

ChristieLaferriereTaxCourt committed:
adding cache-cypress id to retrieve cache if available

ChristieLaferriereTaxCourt committed:
testing the cypress binary path

ChristieLaferriereTaxCourt committed:
restore config.yml file to just 8 parralel runs

ChristieLaferriereTaxCourt committed:
increasing parallelization to 10

ChristieLaferriereTaxCourt committed:
decreasing parallelization to 9

ChristieLaferriereTaxCourt committed:
experimenting with caching again

ChristieLaferriereTaxCourt committed:
comment out cypress binarry install cypress regardless

zacharybrown204 committed:
Cypress performance updates

Removing redundant call of npm ci
Changing hash of package-lock to test cache

zacharybrown204 committed:
Cypress performance tweaks

Updating install condition after a cache miss
Adding extra

ChristieLaferriereTaxCourt committed:
adding second condiiton for clean install if cache miss

zacharybrown204 committed:
Cypress performance tweaks

Reordering conditions on fallback clean install step
Adding continue-on-error to install step so it can move onto the fallback clean install step.

ChristieLaferriereTaxCourt committed:
re-enable cypress cache, upgrade cache version to v3 form v3

ChristieLaferriereTaxCourt committed:
restore cache retrieval for cypress

ChristieLaferriereTaxCourt committed:
check if cypress install isolation allows cypress test to pass

zacharybrown204 committed:
cypress test tweaks

Setting the normal npm install to always run after restoring from cache.

zacharybrown204 committed:
Cypress tweaks:

(temp) Flooding log file will all of the application runtime logs

zacharybrown204 committed:
Cypress tweaks:

Trying to install puppeteer when cache miss

zacharybrown204 committed:
Cypress tweaks

Accidently set puppeteer to install on cache miss instead of hit

ChristieLaferriereTaxCourt commented:
RESTORED AND RESOLVED ✅ 

*‼️‼️‼️‼️‼️‼️‼️*Reminder to us @zacharybrown204 @ChristieLaferriereTaxCourt: REMOVE VERBOSE LOGS and put back npm run start:all:ci >> /tmp/cypress/cypress-output.txt & ./wait-until-services.sh**

zacharybrown204 committed:
Cypress tweaks:

Adding a cache for puppeteer

ChristieLaferriereTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

ChristieLaferriereTaxCourt committed:
clean up

ChristieLaferriereTaxCourt committed:
Merge branch '10478-opex-speed-up-cypress-tests-in-deployed-environmnts' of github.com:ustaxcourt/ef-cms into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

ChristieLaferriereTaxCourt committed:
more clean up

ChristieLaferriereTaxCourt committed:
Merge branch 'staging' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

ChristieLaferriereTaxCourt committed:
install updated dependencies

ChristieLaferriereTaxCourt committed:
match package-lock.json to staging

ChristieLaferriereTaxCourt committed:
Merge branch 'staging' into experimental8

ChristieLaferriereTaxCourt committed:
npm install

ChristieLaferriereTaxCourt committed:
remove newline

ChristieLaferriereTaxCourt committed:
Merge branch 'experimental8' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

TomElliottFlexion commented on package.json:
```diff
@@ -328,6 +329,7 @@
     "lint-staged-shellcheck": "0.1.2",
     "livereload": "0.10.3",
     "nodemon": "3.1.10",
+    "p-limit": "^7.1.1",
```
Tiny nit: this should be pinned

ChristieLaferriereTaxCourt replied:
Thank you for pointing this out! 🙂 I have made the adjustment

ChristieLaferriereTaxCourt committed:
add new line

ChristieLaferriereTaxCourt committed:
add new line

ChristieLaferriereTaxCourt committed:
Merge branch 'experimental8' into 10478-opex-speed-up-cypress-tests-in-deployed-environmnts

ChristieLaferriereTaxCourt committed:
pin the p-limit package

TomElliottFlexion commented on .circleci/config.yml:
```diff
@@ -604,6 +613,10 @@ jobs:
             if [ $MIGRATE_FLAG == true ]; then
               npm run migration:cleanup -- $ENV
             fi
+      - run:
+          name: Cleanup Cypress Test Accounts
+          command: |
+            npm run cleanup:cypress-test-accounts
```
This is a great idea!

TomElliottFlexion replied:
Not a change request, but something to consider: if we clean up cypress test accounts at the end of a Circle step instead of in the tests themselves, individually run smoketests will create orphaned accounts. This is probably worth the tradeoff, especially if everyone knows to call `npm run cleanup:cypress-test-accounts` after running smoketests against a deployed environment, just putting the (minor) risk in writing

jimlerza replied:
We actually discussed this!

Even if we forget to run the cleanup manually (we will _definitely_ forget to run the cleanup manually!) the way it works, it'll clean up everything that needs cleanin'.

So, for better or worse, "the next deployment will get it."

TomElliottFlexion reviewed (approved):
Thank you!

ChristieLaferriereTaxCourt commented:
For anyone reviewing this PR, here is some additional context. This feature branch has the updates to reduce the time it takes for Github Action checks to run to completion and experimental8 branch has the updates to reduce the time to deploy to environments to completion. Here is the last successful build on experimental8 branch that has been merged into this feature branch: [Successful last deploy in exp8 environment](https://app.circleci.com/pipelines/github/ustaxcourt/ef-cms?branch=experimental8.)

Staging had also been back-merged into both this feature branch and exp8 branch before exp8 was merged into this feature branch.

jimlerza commented on cypress/deployed-and-local/integration/caseDetail/docketRecord/courtIssuedFiling/order-gets-unserved.cy.ts:
![gustin meme](https://en.meming.world/images/en/d/dd/Grant_Gustin_Next_To_Oliver_Queen%27s_Grave.jpg)

jimlerza commented on cypress/deployed-and-local/integration/caseDetail/docketRecord/courtIssuedFiling/order-signature-disappears.cy.ts:
🪦

jimlerza reviewed (approved):
🔥 

Crushed it!

jimlerza commented:
I know you wanted to do more documentation, so I didn't delete the feature branch. If you want to make another PR to staging, feel free.

But I just couldn't wait. This is AWESOME! 😄
